### PR TITLE
Accept backend solver parameters

### PR DESCRIPTION
**Pull request by cmangla**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: f1325baf2c5ed8b250cc8649446d767cdbdb5745).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/135551394
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

File: scripts/minisat-2.2.1-export-parseSingleOption.patch [hunk 1]
```diff
@@ -0,0 +1,76 @@
+From f394073844d6abe65dbd81eb6a787ae25e780e77 Mon Sep 17 00:00:00 2001
```
Comment (cmangla) [inline scripts/minisat-2.2.1-export-parseSingleOption.patch]: Patch created from a git commit, better viewed here:
https://github.com/cmangla/minisat-2.2.1-for-cbmc/commit/f394073844d6abe65dbd81eb6a787ae25e780e77?diff=unified

File: src/solvers/sat/satcheck_minisat2.cpp [hunk 2]
```diff
@@ -371,3 +371,11 @@ bool satcheck_minisat_simplifiert::is_eliminated(literalt a) const
 
   return solver->isEliminated(a.var_no());
 }
+
+bool satcheck_minisat_simplifiert::set_parameter(const char *const param)
+{
+  int p_len = strlen(param) + 2;
+  char *p = new char[p_len];
```
Comment (smowton) [inline src/solvers/sat/satcheck_minisat2.cpp]: Is it intentional that this throws away the `char[]` allocation without deletion? If not, suggest using `std::string` or similar to hold and manage the lifespan of the memory.

Comment (cmangla) [reply src/solvers/sat/satcheck_minisat2.cpp]: No, not intentional, it's a mistake. Actually I should be using std::string throughout the changes. Will fix. Thanks for the feedback @smowton !

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 092931428d75f5c63e58dedcb056c33a2a810c33).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/136046742
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 43de44227bf06a4f1973c54b1c3372bb71086cbc).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/136054521
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by smowton — APPROVED** [review]

**Comment by smowton** [comment]
To remedy the codecov failure, add a test to `regression/cbmc` that verifies this (ideally it should also verify rejection of bad parameters)

File: scripts/minisat-2.2.1-export-parseSingleOption.patch [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     friend  void printUsageAndExit (int  argc, char** argv, bool verbose);
+     friend  void setUsageHelp      (const char* str);
+     friend  void setHelpPrefixStr  (const char* str);
++    friend  bool parseSingleOption (const char* option);
+ };
+ 
+ 
+-- 
+2.23.0
+
```
Comment (cmangla) [inline scripts/minisat-2.2.1-export-parseSingleOption.patch:76]: Patch created from a git commit, better viewed here:
cmangla/minisat-2.2.1-for-cbmc@f394073

**Comment by feliperodri** [comment]
I think we can confidently close this PR, but I want to confirm there is no problem with @thomasspriggs @chrisr-diffblue

**Comment by thomasspriggs** [comment]
I am OK with the idea of closing this out unmerged. Unless we have someone who especially wants to tune the parameters to MiniSat. This PR would increase the number of patches we apply to MiniSat, which could potentially increase the maintenance burden. It would also expand the MiniSat API which could increase the work involved in swapping in other forks of MiniSat. Therefore I am in favour of closing this out. @chrisr-diffblue ?

**Comment by chrisr-diffblue** [comment]
Yep, I'll close this one, thanks!

**Comment by chrisr-diffblue** [comment]
The this draft PR appears to be fairly old and outdated, so we'll close it now - but if this is actively being worked on, feel free to reopen.

**Pull request closed without merging**